src/main/java/hyperledger/besu/java/rest/client/config/KafkaConsumerConfig.java

```diff
@@ -0,0 +1,80 @@
+package hyperledger.besu.java.rest.client.config;
+
+import java.util.HashMap;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.common.config.SslConfigs;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.maven.surefire.shade.org.apache.commons.lang3.StringUtils;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.cloud.context.config.annotation.RefreshScope;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+
+/*
+ * This class is the configuration class for setting the properties for the kafka consumers.
+ *
+ */
+@Slf4j
+@Configuration
+@ConditionalOnProperty("kafka.integration-points[0].brokerHost")
+@RefreshScope
+public class KafkaConsumerConfig {
```

Comment:
Can we delegate task grouped by partition to threads in a pool and allow parallel processing. That way, we can speed up the performance of the Consumer by addressing few of the concerns like offset commit (manual), pause/revoke the consumer thread, adding rebalancing logic for consumer if that gets affected etc.

Reply:
Sure, I will work on it. In the initial version, I will keep it same as what exists today for hlf-connector.